easybuild/easyblocks/o/openfoam.py

```diff
@@ -136,12 +136,18 @@ def configure_step(self):
             self.log.debug("List of deps: %s" % self.cfg.dependencies())
             for dep in self.cfg.dependencies():
                 self.cfg.update('prebuildopts', "%s_SYSTEM=1" % dep['name'].upper())
-                self.cfg.update('prebuildopts', "%(name)s_LIB_DIR=$EBROOT%(name)s/lib" % {'name': dep['name'].upper()})
-                self.cfg.update('prebuildopts', "%(name)s_INCLUDE_DIR=$EBROOT%(name)s/include" % {'name': dep['name'].upper()})
+                self.cfg.update('prebuildopts', "{0}_LIB_DIR=$EBROOT{0}/lib".format(dep['name'].upper()))
+                self.cfg.update('prebuildopts', "{0}_INCLUDE_DIR=$EBROOT{0}/include".format(dep['name'].upper()))
         else:
-            scotch = get_software_root('SCOTCH')
-            if scotch:
-                self.cfg.update('prebuildopts', "SCOTCH_ROOT=$EBROOTSCOTCH")
+            for depend in ['SCOTCH', 'METIS', 'CGAL', 'PARAVIEW']:
+                dependloc = None
+                dependloc = get_software_root(depend)
+                if dependloc:
+                    if depend == 'CGAL' and get_software_root('BOOST'):
```

Comment:
@boegel `CGAL` have a `Boost` dependency. When we use `CGAL`, we have to give to `OpenFOAM` where is `Boost`. The original version did not check whether we have `Boost`, because I assumed that is available in `CGAL` (since it depends on it). Now I am checking it (as you requested previously). All in all, we need the extra boost `configopt` only if we have `CGAL`. And If we have `CGAL` then withourt giving the `Boost` location, the compilation is unsuccessful, I assume it needs some `Boost` header files. The current `CGAL` cannot be compiled without `Boost` but I do not know about past and future versions. (all of our `CGAL` easyconfigs have `Boost` dependency)